src/cmd/ksh93/tests/builtins.sh

```diff
@@ -896,6 +896,8 @@ else
     err_exit 'cannot cd to ~{fd} when fd is /dev'
 fi
 
+mkdir $tmp/oldpwd
+[[ $(OLDPWD="$tmp/oldpwd" cd -) == "$tmp/oldpwd" ]] || err_exit "cd - does not recognize overriden OLDPWD variable"
```

Comment:
Thx for adding a test. However, the wording of the error isn't quite correct. In fact, it's probably worth adding a related test:

```
cd /
OLDPWD=$tmp/oldpwd
cd -
[[ $PWD == "$tmp/oldpwd" ]] || err_exit "cd - does not recognize overridden OLDPWD variable"
```

That's because overriding OLDPWD does work but not if done by creating a new env context. The above test should pass without this fix.

Also, s/overriden/overridden/

Reply:
I have added the test case that you suggested.